contracts/gmp-express/GMPExpressService.sol

```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.9;
+
+import { Upgradable } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/upgradable/Upgradable.sol';
+import { SafeTokenTransfer, SafeNativeTransfer } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/utils/SafeTransfer.sol';
+import { IExpressProxy } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IExpressProxy.sol';
+import { IERC20 } from '@axelar-network/axelar-gmp-sdk-solidity/contracts/interfaces/IERC20.sol';
+import { IGMPExpressService } from '../interfaces/IGMPExpressService.sol';
+import { ExpressProxyFactory } from './ExpressProxyFactory.sol';
+
+contract GMPExpressService is Upgradable, ExpressProxyFactory, IGMPExpressService {
```

Comment:
This `GMPExpressService` inherits `ExpressProxyFactory` really confuses me. Why aren't they separate? Why does the service have to worry about deploying it in other places?

Reply:
GMPExpressService is responsible for validating the proxy before performing an express call. And validation comes together with deployment. The purpose of the service is to serve all the purposes related to GMPE.